### PR TITLE
[Storage] Migrate test_container.py

### PR DESCRIPTION
**Pull request by vincenttran-msft**
This PR migrates `test_container.py` and `test_container_async.py` to use test proxy. Still need to address #25623 once the batch sanitizer resolution has been reached.

### CODE REVIEW

File: sdk/storage/azure-storage-blob/tests/test_container.py [hunk 1]
```diff
@@ -1772,7 +2090,10 @@ def test_standard_blob_tier_set_tiers_with_sas(self, storage_account_name, stora
     # once we have premium tests, still we don't want to test Py 2.7
     # @pytest.mark.skipif(sys.version_info < (3, 0), reason="Batch not supported on Python 2.7")
     @BlobPreparer()
-    def test_premium_tier_set_tier_api_batch(self, storage_account_name, storage_account_key):
+    def test_premium_tier_set_tier_api_batch(self, **kwargs):
```
Comment (jalauzon-msft) [inline sdk/storage/azure-storage-blob/tests/test_container.py:2093]: Did you try anything with this test? Looks like we need a Premium Page Blob account that supports batch? That just may not be a thing :)

File: sdk/storage/azure-storage-blob/tests/test_container_async.py [hunk 2]
```diff
@@ -5,64 +5,48 @@
 # Licensed under the MIT License. See License.txt in the project root for
```
Comment (jalauzon-msft) [inline sdk/storage/azure-storage-blob/tests/test_container_async.py:5]: nit: (Can't leave comment on part of file not in diff) but can you remove the `# coding: utf-8` at the top? I don't think we need that anymore and I've been removing it.

File: sdk/storage/azure-storage-blob/tests/conftest.py [hunk 3]
```diff
@@ -13,7 +13,8 @@
     add_header_regex_sanitizer,
     add_oauth_response_sanitizer,
     add_uri_regex_sanitizer,
-    test_proxy
+    test_proxy,
+    add_body_key_sanitizer
```
Comment (jalauzon-msft) [inline sdk/storage/azure-storage-blob/tests/conftest.py:17]: I think you can undo this now?

**Review by jalauzon-msft — APPROVED** [review]
Again, great work on this tough file!

File: sdk/storage/azure-storage-blob/tests/test_container.py (hunk 1 above)
Comment (vincenttran-msft) [reply sdk/storage/azure-storage-blob/tests/test_container.py:2093]: Yeah I have no idea what "premium" means in this context 😆. I remember trying a bunch of different things when initially triaging this in our 'triage all skipped tests' item from a couple weeks / month(?) ago, so just left it skipped for now (it's in the backlog!! 🤣)

File: sdk/storage/azure-storage-blob/tests/test_container_async.py (hunk 2 above)
Comment (vincenttran-msft) [reply sdk/storage/azure-storage-blob/tests/test_container_async.py:3]: That I can do!

File: sdk/storage/azure-storage-blob/tests/conftest.py (hunk 3 above)
Comment (vincenttran-msft) [reply sdk/storage/azure-storage-blob/tests/conftest.py]: Done!

**Pull request merged**